core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -97,10 +97,29 @@
         SNAPSHOT
     }
 
+    /**
+     * Indicates that this custom metadata will be returned as part of API call but will not be persisted
```

Comment:
Thank you for adding these comments!
Nit: part of API call -> part of an API call  (same for the descriptions below)